contracts/RewardPool.sol

```diff
@@ -92,6 +96,24 @@ contract RewardPool is AccessControl, ReentrancyGuard {
     uint256 feeAmount,
     uint256 protocolFee
   );
+  event RewardFunctionAddressUpdated(
+    address newRewardFunctionAddress,
+    address previousRewardFunctionAddress
+  );
+  event AddKpi(
+    address indexed user,
+    bytes32 indexed rewardPeriodKey,
+    uint256 amount,
+    uint256 rewardPeriodStart,
+    uint256 rewardPeriodEndExclusive,
```

Comment:
nit: this is sufficient (or even uint48)
```suggestion
    uint64 rewardPeriodStart,
    uint64 rewardPeriodEndExclusive,
```

OpenZeppelin contracts does that too ([example 1](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/53bb34057ed97ea9b36d550f1c2c413ef5b6c6bb/contracts/access/extensions/AccessControlDefaultAdminRules.sol#L56) or [example 2](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/types/Time.sol)).